api/controller/dbController.js

```diff
@@ -0,0 +1,94 @@
+import { query } from '../../db/db'
+import { 
+  artistQueries, setlistQueries, songQueries, setlistSongQueries
+  } from '../../db/queries'
+
+  export class dbController {
+    artistId;
+    concertId;
+    songId;
+    constructor(apiResult, artist, song, artistId, concertId, songId) {
+      this.apiResult = apiResult;
+      this.artist = artist;
+      this.song = song;
+      this.externalId = apiResult.setlist[0].artist.mbid;
+      this.artistId = artistId;
+      this.concertId = concertId;
+      this.songId = songId;
+    }
+
+    addSong = async (song) => {
+      const addResult = await query(songQueries.addSong,[song,this.artistId]);
+      this.songId = addResult[0].id;
+    }
+    
+    getSong = async (song) => {
+      const getResult = await query(songQueries.getSong,[song,this.artistId]);
+      const songInDb = getResult.length > 0;
+      if (songInDb) {
+        this.songId = getResult[0].id;
+      } else {
+        this.songId = null;
+      }
+    }
+
+    addSetlistSong = async () => {
+      await query(setlistSongQueries.addSetlistSong,[this.concertId,this.songId])
+    }
+
+    addSetlist = async (externalId,updatedAt,eventDate) => {
+      const result = await query(setlistQueries.addSetlist,[externalId,this.artistId,updatedAt,eventDate]);
+      return result[0].id;
+    }
+  
+    addedArtist = async () => {
+      const result = await query(artistQueries.addArtist,[this.externalId,this.artist]);
+      this.artistId = result[0].id;
+    }
+  
+    getArtist = async () => {
+      const result = await query(artistQueries.getArtist,[this.artist]);
+      this.artistId = result.length > [0] ? result[0].id : 0;
+      return this.artistId;
+    }
+
+    getDates = async () => {
+      const returnedValue = {};
+      const result = await query(setlistSongQueries.getSetlistSong,[this.song,this.artistId])
```

Comment:
I would actually argue that the query method `getSetlistSong` is named incorrectly.

`getSetlistSong` _does_ seem deceptive, the reason being is you aren't actually getting the song, you're getting the dates of the setlist songs (or performances) for a given song/artist combo. Something more like `getSongPerformanceDates`. We can talk about this one more in person, there might be other solutions that are more systemic and less obvious.